db/migrate/20200210150200_associate_move_with_profile_part2.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class AssociateMoveWithProfilePart2 < ActiveRecord::Migration[5.2]
+  def up
+    Move.find_each.reject { |m| m.profile_id.present? }.each do |move|
+      move.profile_id = Person.find_by!(id: move.person_id).latest_profile.id
+      move.save!
+    end
+
+    change_column_null :moves, :profile_id, false
+    add_foreign_key :moves, :profiles
+
+    change_table :moves do |t|
+      t.index [:from_location_id, :to_location_id, :profile_id, :date], name: "index_move_loc_profile_date", unique: true
+    end
+
+    # person_id is unused after this migration, so it has to be nullable
+    change_column_null(:moves, :person_id, true)
+    remove_foreign_key :moves, :people
+  end
+
+  def down
+    remove_foreign_key :moves, :profiles
+    change_column_null :moves, :profile_id, true
+
+    Move.find_each do |move|
```

Comment:
Should probably just be setting the person_id for those records which don't have one, i.e.
```
Move.where(person_id: nil).each do |move|
```

Reply:
again see above. Yes this could be marginally more efficient, but the code you suggest will instantiate every Move object in the DB.

Reply:
see above as a compromise: 

    Move.where(person_id: nil).find_each do |move|